AIPscan/Data/data.py

```diff
@@ -292,3 +299,130 @@ def largest_files(storage_service_id, file_type=None, limit=20):
         report[FIELD_FILES].append(file_info)
 
     return report
+
+
+def _query_aips_by_file_format_or_puid(
+    storage_service_id, search_string, original_files=True, file_format=True
+):
+    """Fetch information on all AIPs with given format or PUID from db.
+
+    :param storage_service_id: Storage Service ID (int)
+    :param search_string: File format or PUID (str)
+    :param original_files: Boolean indicating whether returned data
+    describes original (default) or preservation files.
+    :param file_format: Boolean indicating whether to filter on file
+    format (default) or PUID.
+
+    :returns: SQLAlchemy query results
+    """
+    AIP_ID = "id"
+    TRANSFER_NAME = "name"
+    AIP_UUID = "uuid"
+    FILE_COUNT = "file_count"
+    FILE_SIZE = "total_size"
+    aips = (
+        db.session.query(
+            AIP.id.label(AIP_ID),
+            AIP.transfer_name.label(TRANSFER_NAME),
+            AIP.uuid.label(AIP_UUID),
+            db.func.count(File.id).label(FILE_COUNT),
+            db.func.sum(File.size).label(FILE_SIZE),
+        )
+        .join(File)
+        .join(StorageService)
+        .filter(StorageService.id == storage_service_id)
+        .group_by(AIP.id)
+        .order_by(db.func.count(File.id).desc(), db.func.sum(File.size).desc())
+    )
+
+    if original_files is False:
+        aips = aips.filter(File.file_type == FileType.preservation.value)
+    else:
+        aips = aips.filter(File.file_type == FileType.original.value)
+
+    if file_format:
+        return aips.filter(File.file_format == search_string)
+    return aips.filter(File.puid == search_string)
+
+
+def _aips_by_file_format_or_puid(
+    storage_service_id, search_string, original_files=True, file_format=True
+):
+    """Return overview of all AIPs containing original files in format
+
+    :param storage_service_id: Storage Service ID (int)
+    :param search_string: File format name or PUID (str)
+    :param original_files: Boolean indicating whether returned data
+    describes original (default) or preservation files.
+    :param file_format: Boolean indicating whether to filter on file
+    format or PUID (defaults to True).
+
+    :returns: "report" dict containing following fields:
+        report["StorageName"]: Name of Storage Service queried
+        report["AIPs"]: List of result AIPs ordered desc by count
+    """
+    report = {}
+
+    storage_service = _get_storage_service(storage_service_id)
+    report[FIELD_STORAGE_NAME] = ""
+    try:
+        report[FIELD_STORAGE_NAME] = storage_service.name
+    except AttributeError:
+        pass
```

Comment:
You know, you make a good point. I can't remember why I put in this try/except block now but you're right that it doesn't seem likely, so I will change this.

The question about using the ID rather than the name is interesting. I'm tempted to stick with the name just because the Storage Service ID is rather internal to AIPscan. Another thing we could consider using moving forward is the Storage Service URL, but I'm not sure if including that in reports would have security implications.

If it's alright with you, I will revert this to `report[FIELD_STORAGE_NAME] = storage_service.name` for this PR, and maybe we can keep pondering how to best represent Storage Service information in the reports separately as the need arises.

Reply:
Sounds good to me, and great points about that.